### DISCUSSION TITLE
Preparing for 3.0

### DISCUSSION THREAD

**Discussion by NOTiFYcouk** [MorphiaOrg/morphia#2528] [General]
"Could not resolve dev.morphia.morphia:morphia-core:3.0.0-SNAPSHOT"

Can only see:

https://mvnrepository.com/artifact/dev.morphia.morphia/morphia-core

```
> Task :compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Could not resolve all files for configuration ':compileClasspath'.
   > Could not find dev.morphia.morphia:morphia-core:3.0.0-SNAPSHOT.
     Searched in the following locations:
       - file:/Users/NOTiFY/.m2/repository/dev/morphia/morphia/morphia-core/3.0.0-SNAPSHOT/maven-metadata.xml
       - file:/Users/NOTiFY/.m2/repository/dev/morphia/morphia/morphia-core/3.0.0-SNAPSHOT/morphia-core-3.0.0-SNAPSHOT.pom
       - https://repo.maven.apache.org/maven2/dev/morphia/morphia/morphia-core/3.0.0-SNAPSHOT/maven-metadata.xml
       - https://repo.maven.apache.org/maven2/dev/morphia/morphia/morphia-core/3.0.0-SNAPSHOT/morphia-core-3.0.0-SNAPSHOT.pom
       - https://dl.google.com/dl/android/maven2/dev/morphia/morphia/morphia-core/3.0.0-SNAPSHOT/maven-metadata.xml
       - https://dl.google.com/dl/android/maven2/dev/morphia/morphia/morphia-core/3.0.0-SNAPSHOT/morphia-core-3.0.0-SNAPSHOT.pom
       - https://repository.primefaces.org/dev/morphia/morphia/morphia-core/3.0.0-SNAPSHOT/maven-metadata.xml
       - https://repository.primefaces.org/dev/morphia/morphia/morphia-core/3.0.0-SNAPSHOT/morphia-core-3.0.0-SNAPSHOT.pom
       - https://plugins.gradle.org/m2/dev/morphia/morphia/morphia-core/3.0.0-SNAPSHOT/maven-metadata.xml
       - https://plugins.gradle.org/m2/dev/morphia/morphia/morphia-core/3.0.0-SNAPSHOT/morphia-core-3.0.0-SNAPSHOT.pom
       - file:/Users/NOTiFY/IdeaProjects/NOTiFYmoto/lib/morphia-core-3.0.0-SNAPSHOT.jar
       - file:/Users/NOTiFY/IdeaProjects/NOTiFYmoto/lib/morphia-core.jar
     Required by:
         project :
```

Where can I find 3.0.0-SNAPSHOT etc. with the JAR? TIA

**Comment by evanchooly** [comment]
Did you add the [snapshot repo](https://morphia.dev/morphia/3.0/index.html) to your build?

**Reply by NOTiFYcouk** [reply]
`earlib group: "dev.morphia.morphia", name: "morphia-core", version: "3.0.0-SNAPSHOT", ext: "jar"
`

**Reply by evanchooly** [reply]
but did you add the snapshot repo to your build?  the snapshot artifacts don't live on maven central.

**Comment by NOTiFYcouk** [comment]
I aways have on:

`MyApp/lib/morphia-core-2.4.2.jar`

Where can I find the "3.0.0-SNAPSHOT.jar" ?

**Reply by evanchooly** [reply]
if you're using maven or gradle, you need to update your build.   if you're manually downloading jars, the snapshot jars are [here](https://oss.sonatype.org/content/repositories/snapshots/dev/morphia/morphia/morphia-core/).

**Comment by NOTiFYcouk** [comment]
**Fixed. Running fine on 3.0.0-x**. TIA.

FYI:

Gradle 8.2:

```
implementation("dev.morphia.morphia:morphia-core:3.0.0-20230725.035612-1")
...
earlib group: "dev.morphia.morphia", name: "morphia-core", version: "3.0.0-20230725.035612-1", ext: "jar"
```

MyApp: Sample config files -> Complete with defaults

`META-INF/morphia-config.properties`

With:

```
earlib group: "org.mongodb", name: "mongodb-driver-core", version: "4.10.2", ext: "jar"
earlib group: "org.mongodb", name: "mongodb-driver-sync", version: "4.10.2", ext: "jar"
earlib group: "org.mongodb", name: "bson", version: "4.10.2", ext: "jar"
```

EAR deployed: 

```
JBOSS_HOME: /usr/local/Cellar/wildfly-as/wildfly-preview-29.0.1.Final
...
JAVA: /Library/Java/JavaVirtualMachines/jdk-20.0.2.jdk/Contents/Home/bin/java

```
INFO  [com.notifymoto.controller.MotorcycleController] (default task-1) >>>>> MotorcycleController selectedMotorcycle = Moto{id=6409ecf6305f64d3e635a0b7, manufacturer='Moto Guzzi', model='V85', modelType='TT Evocative Graphics', typesOf='Enduro', dateProductionStarted=Sun Jan 01 00:00:00 GMT 2023, dateProductionEnded=Sun Jan 01 00:00:00 GMT 2023, engine=Engine{type='Four-Stroke', displacement=0.0, cylinder=2.0, capacityUnit='cc', bore=0.0, boreMeasurement='mm', stroke=0.0, strokeMeasurement='mm', distribution='null', maxiumPowerHp=0.0, maxiumPowerKilowatt=0.0, maxiumPowerRpm=0.0, maximumTorque=0.0, maximumTorqueUnit='Nm', maximumTorqueRpm=0.0}, performance=null, chassisBrakesSuspensionWheels=null, dimensionsWeights=null, transmission=null, instruments=null, electrics=Electrics{ignition='Electrics Ignition'}, colours=[Colour{colour='n/a'}], accessories=null, image=Image{file='/Users/NOTiFY/IdeaProjects/NOTiFYmoto/images/moto', url='/MotoGuzzi/2023/', png='gu9032274eum02-01-m.webp', dimensionsWidth='1500', dimensionsHeight='1000'}, imageList=[Image{file='/Users/NOTiFY/IdeaProjects/NOTiFYmoto/images/moto', url='/MotoGuzzi/2023/', png='gu9032274eum02-01-m.webp', dimensionsWidth='1500', dimensionsHeight='1000'}]}
